packages/backend/src/models/SavedChartModel.ts

```diff
@@ -180,44 +157,47 @@ const createSavedChartVersion = async (
                 timezone,
             })
             .returning('*');
-        const promises: Promise<any>[] = [];
-        dimensions.forEach((dimension) => {
-            promises.push(
-                createSavedChartVersionField(trx, {
+        if (dimensions.length > 0) {
```

Comment:
nice, you could also check `if (dimensions.length > 0)` inside `createSavedChartVersionFields`

Reply:
yup, thats cleaner and reusable. Moved those checks inside the functions and triggered frontend tests to be sure nothing breaks.